libraries/tickfs/src/flash_controller.rs

```diff
@@ -0,0 +1,79 @@
+//! The Flash Controller interface with hardware
+
+use crate::error_codes::ErrorCode;
+
+/// Implementation required for the flash controller hardware. This
+/// should read, write and erase flash from the hardware using the
+/// flash controller.
+///
+/// This is the public trait for the Flash controller implementation.
+///
+/// The PageSize 'P' must be the smallest size that can be erased in
+/// a single operation.
+/// The start and end address of the FlashController must be aligned
+/// to the PageSize.
+/// All `region_number`s and `address`es are offset from zero. If you
```

Comment:
How do an implementer and user agree on how large a region is?

Reply:
I have updated the documentation to make this clearer.

Reply:
What about making `get_region_size()` a member function in the trait (or maybe some other rust-y approach)?

Reply:
With const generic arrays I think this would be a little clearer. I'll see if I can find a rust-y way to make this nicer.